Peaky.Tests/TestClasses/ParametrizedTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentAssertions;
+
+namespace Peaky.Tests.TestClasses
+{
+    public class ParametrizedTest : IApplyToApplication, IParametrizedTestCases
+    {
+        private readonly Dictionary<string, EnvironmentStuff> _environmentLookup;
+        public bool AppliesToApplication(string application) => application.Equals("parametrized", StringComparison.OrdinalIgnoreCase);
+
+        public ParametrizedTest()
+        {
+            _environmentLookup = new Dictionary<string, EnvironmentStuff>
+            {
+                { "case1", new EnvironmentStuff{Value = true} },
+                { "case2", new EnvironmentStuff {Value = false} },
+                { "case3", new EnvironmentStuff {Value = true} },
+                { "case4", new EnvironmentStuff{Value = false} }
+            };
+        }
+
+        public void I_do_stuff(string testCaseId, bool extectedResult)
+        {
+            var env = _environmentLookup[testCaseId];
+            env.Value.Should().Be(extectedResult);
+        }
+
+        public bool I_do_stuff_and_return(string testCaseId, bool extectedResult)
```

Comment:
I would rename to e.g. `I_do_stuff_and_return_bool` and then add a couple of other cases that are often interesting:

* `I_do_stuff_and_return_Task` 
* `I_do_stuff_and_return_Task_of_T`